src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/AnnotationMetadataStepTest.java

```diff
@@ -0,0 +1,138 @@
+import uk.ac.ebi.eva.pipeline.jobs.AnnotationJob;
+import uk.ac.ebi.eva.pipeline.parameters.MongoConnection;
+import uk.ac.ebi.eva.test.configuration.BatchTestConfiguration;
+import uk.ac.ebi.eva.test.rules.PipelineTemporaryFolderRule;
+import uk.ac.ebi.eva.test.rules.TemporaryMongoRule;
+import uk.ac.ebi.eva.utils.EvaJobParameterBuilder;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * TODO jmmut remove import AnnotationJob when we add the stepLauncherTestUtils
+ */
+@RunWith(SpringRunner.class)
+@ActiveProfiles(Application.VARIANT_ANNOTATION_MONGO_PROFILE)
+@TestPropertySource({"classpath:common-configuration.properties", "classpath:test-mongo.properties"})
+@ContextConfiguration(classes = {AnnotationJob.class, BatchTestConfiguration.class})
+public class AnnotationMetadataStepTest {
+    @Rule
+    public TemporaryMongoRule mongoRule = new TemporaryMongoRule();
+
+    @Rule
+    public PipelineTemporaryFolderRule temporaryFolderRule = new PipelineTemporaryFolderRule();
+
+    @Autowired
+    private JobLauncherTestUtils jobLauncherTestUtils;
+
+    @Autowired
+    private MongoConnection mongoConnection;
+
+    @Autowired
+    private MongoMappingContext mongoMappingContext;
+
+    @Test
+    public void shouldWriteVersions() throws Exception {
+        String databaseName = mongoRule.getRandomTemporaryDatabaseName();
+        MongoOperations mongoOperations = MongoConfiguration.getMongoOperations(databaseName, mongoConnection,
+                mongoMappingContext);
+        String vepCacheVersion = "87";
+        String vepVersion = "88";
+
+        assertStepCompletes(databaseName, vepCacheVersion, vepVersion);
+
+        //check that the document was written in mongo
+        List<AnnotationMetadata> annotationMetadatas = mongoOperations.findAll(AnnotationMetadata.class);
+
+        assertEquals(1, annotationMetadatas.size());
+        assertEquals(vepCacheVersion, annotationMetadatas.get(0).getCacheVersion());
+        assertEquals(vepVersion, annotationMetadatas.get(0).getVepVersion());
+    }
+
+    private void assertStepCompletes(String databaseName, String vepCacheVersion, String vepVersion) {
+        String collectionAnnotationMetadataName = "annotationMetadata";
+        JobParameters jobParameters = new EvaJobParameterBuilder()
+                .collectionAnnotationMetadataName(collectionAnnotationMetadataName)
+                .databaseName(databaseName)
+                .vepCacheVersion(vepCacheVersion)
+                .vepVersion(vepVersion)
+                .toJobParameters();
+
+        JobExecution jobExecution = jobLauncherTestUtils.launchStep(BeanNames.LOAD_ANNOTATION_METADATA_STEP, jobParameters);
+
+        assertEquals(ExitStatus.COMPLETED, jobExecution.getExitStatus());
+        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
+    }
+
+    @Test
+    public void shouldNotAddRedundantVersions() throws Exception {
```

Comment:
I suspect the method names are swapped between `shouldNotAddRedundantVersions` and `shouldKeepOtherVersions`, as the former currently inserts 2 different versions and the first tries to insert the same twice. It should be the opposite.